x-pack/plugins/synthetics/public/apps/synthetics/components/settings/global_params/add_param_flyout.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiButton,
+  EuiTitle,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormProvider } from 'react-hook-form';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { FETCH_STATUS, useFetcher } from '@kbn/observability-plugin/public';
+import { SavedObject } from '@kbn/core-saved-objects-common';
+import { i18n } from '@kbn/i18n';
+import { syntheticsParamType } from '../../../../../../common/types/saved_objects';
+import { useSyntheticsRefreshContext } from '../../../contexts';
+import { SyntheticsParam } from '../../../../../../common/runtime_types';
+import { useFormWrapped } from '../../../../../hooks/use_form_wrapped';
+import { AddParamForm } from './add_param_form';
+
+export const AddParamFlyout = ({
+  items,
+  isEditingItem,
+  setIsEditingItem,
+  setRefreshList,
+}: {
+  items: Array<SavedObject<SyntheticsParam>>;
+  setRefreshList: React.Dispatch<React.SetStateAction<number>>;
+  isEditingItem: SavedObject<SyntheticsParam> | null;
+  setIsEditingItem: React.Dispatch<React.SetStateAction<SavedObject<SyntheticsParam> | null>>;
+}) => {
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+
+  const form = useFormWrapped<SyntheticsParam>({
+    mode: 'onSubmit',
+    reValidateMode: 'onChange',
+    shouldFocusError: true,
+    defaultValues: isEditingItem?.attributes ?? {
+      key: '',
+      tags: [],
+      description: '',
+      value: '',
+    },
+  });
+
+  const closeFlyout = useCallback(() => {
+    setIsFlyoutVisible(false);
+    setIsEditingItem(null);
+    form.reset({ key: '', tags: [], description: '', value: '' });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you include a comment on why you're excluding exhaustive deps? Excluding exhaustive deps can cause hard to debug bugs, especially when a new dependency is added that is needed but a dev forgets to add and linting doesn't catch it because of the disabling. This has happened to Justin and I more than once.